### PR TITLE
Update swift lint and format + appy fixes

### PR DESCRIPTION
**Pull request by mat1th**
## Summary
Swift lint and swiftformat are outdated. This PR does update those + applies the new formatting form swiftformat. 
There is 1 swift file with a manual change: `Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift`. This is done because `swiftlint` did create the following swiftlint error: `error: Cyclomatic Complexity Violation: Function should have complexity 10 or less; currently complexity is 11 (cyclomatic_complexity)`.

Because it does change a lot of files the question is if we want to finetune the `swiftformat` rules.

## Screenshots
No user facing changes.

## Link to pull request in Documentation repository
NA

## Any other notes
NA

### CODE REVIEW

**Comment by bgoncal** [comment]
Could you check tests? And you got a conflict

**Comment by mat1th** [comment]
> Could you check tests? And you got a conflict

The tests are broken because of the swift version upgrade to fix this I've opend https://github.com/home-assistant/iOS/pull/2593 to not intruduce to many changes in this pr. So that pr needs to be merged before this one.

**Comment by mat1th** [comment]
The tests are green again, after the update from master.

File: Sources/App/Scenes/SceneManager.swift [hunk 1]
```diff
@@ -57,7 +57,7 @@ class SceneManager {
             }
         }
 
-        func resolve<T>(with possible: T) {
```
Comment (bgoncal) [inline Sources/App/Scenes/SceneManager.swift:60]: Was this done by SwiftLint/swiftformat?

File: Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift [hunk 2]
```diff
@@ -82,42 +82,14 @@ final class CarPlayAreasViewModel {
...
-                    entityIds.append(entityId)
-                    areasAndEntitiesDict[areaId] = entityIds
-                } else {
-                    areasAndEntitiesDict[areaId] = [entityId]
-                }
-            }
-        }
-
+        let areasAndDevicesDict = mapToareasAndEntities(devicesAndAreas: devicesAndAreas)
+        var areasAndEntitiesDict = mapToAreasAndEntitiesDict(areasAndEntities: areasAndEntities)
```
Comment (bgoncal) [inline Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift]: Have you checked if the behavior is still the same? It looks good, but worth checking. If you dont see carplay right away you can manually add `com.apple.developer.carplay-driving-task` entitlement to test it out (just dont commit it)

**Review by bgoncal — CHANGES_REQUESTED** [review]
Besides CarPlay, any other manual changes?

File: Sources/App/Scenes/SceneManager.swift (hunk 1 above)
Comment (mat1th) [reply Sources/App/Scenes/SceneManager.swift:60]: Yes, it was done by swiftformat. I can create a rule that does not do this change.

File: Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift (hunk 2 above)
Comment (mat1th) [reply Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift]: Are you okay with rewriting this class so it is easier testable, so I can create tests for it? Than we are sure that the before and after are the same.

Comment (bgoncal) [reply Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift]: Sure, that would be really good

Comment (bgoncal) [reply Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift]: If you prefer you can do it on a separate PR and add //swiftlint:disable cyclomatic_complexity for now in this class, so we can keep the PR topic in focus

Comment (mat1th) [reply Sources/Vehicle/Templates/Areas/CarPlayAreasViewModel.swift]: I've reverted the change and disabled the swiftlint rule.

**Comment by mat1th** [comment]
> Besides CarPlay, any other manual changes?

No, all swift file changes are done by `swiftformat` or `swiftlint --fix`.

**Review by bgoncal — APPROVED** [review]
LGTM!

**Pull request merged**